hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncProcess.java

```diff
@@ -218,6 +221,14 @@ public <CResult> AsyncRequestFuture submit(AsyncProcessTask<CResult> task)
     }
   }
 
+  public Map<String, byte[]> getRequestAttributes() {
+    return requestAttributes;
+  }
+
+  public void setRequestAttributes(Map<String, byte[]> requestAttributes) {
+    this.requestAttributes = requestAttributes;
+  }
```

Comment:
This is the part of this changeset that I like least. Ideally the AsyncProcess wouldn't be mutable like this, but because of the abnormal way in which multigets reuse an already constructed AsyncProcess I think this is necessary

Reply:
This is not thread safe though. Every HTable shares the same AsyncProcess. I think you'd add it to the AsyncProcessTask

Reply:
oh yeah, good point. Ok I'll make this change